probably/pgcl/parser.py

```diff
@@ -19,53 +19,76 @@
+from typing import List, Optional, Callable
 
 from lark import Lark, Tree
 
+from probably.pgcl.syntax import has_variable
 from probably.pgcl.ast import *
 from probably.pgcl.walk import Walk, walk_expr
 from probably.util.lark_expr_parser import (atom, build_expr_parser, infixl,
-                                            prefix)
+                                            prefix, infixr)
 from probably.util.ref import Mut
 
 _PGCL_GRAMMAR = """
-    start: declarations instructions
+    start: declarations instructions queries
 
     declarations: declaration* -> declarations
 
     declaration: "bool" var                  -> bool
                | "nat" var bounds?           -> nat
+               | "real" var bounds?          -> real
                | "const" var ":=" expression -> const
+               | "rparam" var                -> rparam
+               | "nparam" var                -> nparam
 
     bounds: "[" expression "," expression "]"
 
     instructions: instruction* -> instructions
 
+    queries: query* -> queries
+
     instruction: "skip"                                      -> skip
                | "while" "(" expression ")" block            -> while
                | "if" "(" expression ")" block "else"? block -> if
                | var ":=" rvalue                             -> assign
                | block "[" expression "]" block              -> choice
                | "tick" "(" expression ")"                   -> tick
+               | "observe" "(" expression ")"                -> observe
+               | "loop" "(" INT ")" block                    -> loop
+
+    query: "?Ex" "[" expression "]"                          -> expectation
+               | "?Pr" "[" expression "]"                    -> prquery
+               | "!Print"                                    -> print
+               | "!Plot" "[" var ("," var)? ("," literal)?"]"-> plot
+               | "?Opt" "[" expression "," var "," var "]"   -> optimize
 
 
     block: "{" instruction* "}"
 
-    rvalue: "unif" "(" expression "," expression ")" -> uniform
+    rvalue: "unif_d" "(" expression "," expression ")" -> duniform
+          | "unif" "(" expression "," expression ")" -> duniform
+          | "unif_c" "(" expression "," expression ")" -> cuniform
+          | "geometric" "(" expression ")" -> geometric
+          | "poisson" "(" expression ")" -> poisson
+          | "logdist" "(" expression ")" -> logdist
+          | "binomial" "(" expression "," expression ")" -> binomial
+          | "bernoulli" "(" expression ")" -> bernoulli
+          | "iid" "(" rvalue "," var ")" -> iid
           | expression
 
     literal: "true"  -> true
            | "false" -> false
            | INT     -> nat
-           | FLOAT   -> float
+           | FLOAT   -> real
            | "∞"     -> infinity
            | "\\infty" -> infinity
 
     var: CNAME
 
+
     %ignore /#.*$/m
-    %ignore /\\/\\/.*$/m
+    %ignore /\/\/.*$/m
```

Comment:
What does this change do?

Reply:
I genuinely don't know, I just copied the new grammar from the forward branch.

Reply:
I think it has something to do with the fact that on the forward branch, this grammar is in a file, which means the `\` etc. don't need to be escaped. I'll revert this to what is on master.

Reply:
Okay, do we have unit tests for comments with the `// comment` syntax? Maybe they work now or are now broken.

Reply:
I'll add one